### ISSUE TITLE
HTML entities might confuse de compiler

### ISSUE THREAD

**Issue by amoutonbrady** [ryansolid/solid#129]
Hey,

I was toying with some random idea and bumped into a somewhat interesting console warning regarding template mismatch. Thankfully, the message was pretty clear and I found the cause rather quickly.

It appears that some HTML entities are causing the compiler to believe the rendered entity is different from the non rendered one (_eg: `&ndash;` is different from `–`_)

Here is an image of that specific warning:

![image](https://user-images.githubusercontent.com/17355226/77226979-8c2b8300-6b7c-11ea-9cde-3a5e24ca90db.png)

And [a small CodeSandbox repro. link](https://codesandbox.io/s/solid-js-entity-template-mismatch-issue-6b117?expanddevtools=1&fontsize=14&hidenavigation=1&moduleview=1&theme=dark)

**Note:** that I tried using the latest beta version 0.17 of solid & solid-preset and then downgraded to the 0.16.14. I reproduced in both cases.
Additionally I'm also using the alpha version of Parcel 2 (_which btw produce amazing result with the latest beta version, about 8kb minified and 3kb gziped for that CodeSandbox_). I doubt it's causing the issue, but if it does then I apologize.

Love your work a lot.

**Comment by ryansolid** [comment]
It isn't. This is a mismatch between the strings made from JSX in the compiler and what the browser outputs after you `innerHTML` it. I started checking after I realize browsers add missing elements like `tbody` which screws up the DOM walker since the precompiled code expects it to be as written. Text matters not at all but structural elements do, but I couldn't think of a cheap non-library bloating way to handle this. Every browser is a bit different and even certain special characters are handled differently. I started using raw to support `<` and `>` which stays encoded where I guess `&nbsp` or `&ndash` don't. You can try this yourself in the console. Notice the difference between them:
```js
d.innerHTML = "<span>&ndash;</span>";
d.innerHTML; //<span>-</span>

d.innerHTML = "<span>&lt;</span>";
d.innerHTML; //<span>&lt;</span>
```
I suppose I could reencode certain characters but not others but unclear if every browser is the same.

Maybe there is a different check I could do like count the number of DOM nodes in the template. Anyone have any ideas?

**Comment by amoutonbrady** [comment]
Ah, I see, that's kind of odd for the browser to that. I reckon, this is quite low level and I'm not sure I understand the subtleties of it. For all I know, similar dom libraries could have the same problem.

For the record, this is not at all blocking on my part and the work around is pretty easy. I just stumbled upon it and thought that you might like to know about it as it didn't seem "normal" to me.

I personally wouldn't like to add extra weight to the library if it's just just a warning and that it doesn't break the application (as it seems to be the case as of now).

Thanks for the clarification :)

**Comment by ryansolid** [comment]
Maybe I could just count the "<"s and if the number is not equal show the message. Wouldn't add much I figure and while isnt full proof is good most of the time.

**Comment by amoutonbrady** [comment]
Had a little bit of time to update the sandbox with the latest beta you released this morning. Totally fixed the issue, congratulation! 🎉 

Peaking at the code, it didn't add that much too, which is always nice.

Thank you Ryan for your reactivity, _pun intended_ 😅

**Comment by ryansolid** [comment]
Great I will consider this closed. And it will be released officially with 0.17.0.
